plugins/metrics/pkg/agent/drivers/prometheus_operator/external_operator.go

```diff
@@ -117,20 +123,29 @@ BACKOFF:
 	return nil
 }
 
-func (d *ExternalPromOperatorDriver) buildPrometheus(conf *node.PrometheusSpec) *monitoringcoreosv1.Prometheus {
+func (d *ExternalPromOperatorDriver) buildPrometheus() *monitoringcoreosv1.Prometheus {
+	return &monitoringcoreosv1.Prometheus{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "opni-prometheus-agent",
+			Namespace: d.Namespace,
+		},
+	}
+}
+
+func (d *ExternalPromOperatorDriver) buildPrometheusAgent(conf *node.PrometheusSpec) *monitoringcoreosv1alpha1.PrometheusAgent {
 	image := "quay.io/prometheus/prometheus:latest"
 	if conf.GetImage() != "" {
 		image = conf.GetImage()
 	}
 
 	selector := &metav1.LabelSelector{}
 
-	return &monitoringcoreosv1.Prometheus{
+	return &monitoringcoreosv1alpha1.PrometheusAgent{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "opni-prometheus-agent",
 			Namespace: d.Namespace,
 		},
-		Spec: monitoringcoreosv1.PrometheusSpec{
+		Spec: monitoringcoreosv1alpha1.PrometheusAgentSpec{
 			CommonPrometheusFields: monitoringcoreosv1.CommonPrometheusFields{
 				Image: &image,
 				Containers: []corev1.Container{
```

Comment:
If we keep anything we have to pretty much keep everything because the operator does not do a merge, however there is a log level field in the actual CRD

Reply:
Yeah sorry I was referring to the LogLevel field in the CRD